gather

```diff
@@ -49,41 +39,86 @@ def run(options=None):
 
 
+    # Import the gatherer(s).
+    sources = options["_"][0].split(",")
+
     # Open CSV writer.
-    gathered_filename = "%s/%s.csv" % (utils.results_dir(), source)
-    gathered_file = open(gathered_filename, 'w', newline='')
-    gathered_writer = csv.writer(gathered_file)
-    gathered_writer.writerow(["Domain", "Base Domain"])
+    if len(sources) > 1:
+        filename = "gathered"
+    else:
+        filename = sources[0] # backwards compatibility, mostly
 
     # De-duping hostnames. This will cause the system to hold all
     # hostnames in memory at once, but oh well.
     hostnames_cache = {}
 
-    # Iterate over each hostname.
-    for domain in gatherer.gather(suffix, options):
-        # Always apply the suffix to returned names.
-        if not suffix_pattern.search(domain):
-            continue
-
-        base = utils.base_domain_for(domain)
+    for source in sources:
+        extra = {}
 
-        # Unless --include-parents is specified, exclude them.
-        if not include_parents:
-            if (domain == base) or (domain == "www.%s" % base):
+        try:
+            gatherer = importlib.import_module("gatherers.%s" % source)
+        except ImportError:
+            # If it's not a registered module, allow it to be "hot registered"
+            # as long as the user gave us a flag with that name that can be
+            # used as the --url option to the URL module.
+            if options.get(source):
+                gatherer = importlib.import_module("gatherers.url")
+                extra['name'] = source
+            else:
+                exc_type, exc_value, exc_traceback = sys.exc_info()
+                logging.error("[%s] Gatherer not found, or had an error during loading.\n\tERROR: %s\n\t%s" % (source, exc_type, exc_value))
+                exit(1)
+
+        # Iterate over each hostname.
+        for domain in gatherer.gather(suffix, options, extra):
+
+            # Always apply the suffix to returned names.
+            if not suffix_pattern.search(domain):
                 continue
 
-        # Apply --parent domain whitelist, if present.
-        if parents:
-            if base not in parents:
-                continue
+            base = utils.base_domain_for(domain)
+
+            # Unless --include-parents is specified, exclude them.
+            if not include_parents:
+                if (domain == base) or (domain == "www.%s" % base):
+                    continue
+
+            # Apply --parent domain whitelist, if present.
+            if parents:
+                if base not in parents:
+                    continue
 
-        # Use hostname cache to de-dupe, if seen before.
-        if domain not in hostnames_cache:
+            # Use hostname cache to de-dupe, if seen before.
+            if domain not in hostnames_cache:
+                hostnames_cache[domain] = [source]
+            elif source not in hostnames_cache[domain]:
+                hostnames_cache[domain] += [source]
```

Comment:
The `hostnames_cache` dict is now not only used to dedupe which hostnames appear, but also to track which sources each hostname does and doesn't appear in.